import-csv.php

```diff
@@ -0,0 +1,91 @@
+<?php
+	require_once "classes/util/View.class.php";
+	require_once "classes/models/Import.class.php";
+	require_once "classes/util/Database.class.php";
+
+	session_start();
+
+	$database = new Database();
+	$db = $database->getNewConnection();
+
+	if (!$db) {
+	    $view = new View("database_error");
+	    $view->send();
+	}
+
+	$import = new Import($db);
+
+	try {
+		if (!isset($_FILES['file-upload']['error']) || is_array($_FILES['file-upload']['error'])) {
+			throw new RuntimeException('Invalid parameters.');
+		}
+
+	    switch ($_FILES['file-upload']['error']) {
+	        case UPLOAD_ERR_OK:
+	            break;
+	        case UPLOAD_ERR_NO_FILE:
+	            throw new RuntimeException('No file sent.');
+	        case UPLOAD_ERR_INI_SIZE:
+	        case UPLOAD_ERR_FORM_SIZE:
+	            throw new RuntimeException('Exceeded filesize limit.');
+	        default:
+	            throw new RuntimeException('Unknown errors.');
+		}
+
+	    if ($_FILES['file-upload']['size'] > 1000000) {
+	    	throw new RuntimeException('Exceeded filesize limit.');
+		}
+
+		# Check file MIME type
+		# Array of valid MIME types
+		$csvMimes = array('text/x-comma-separated-values', 'text/comma-separated-values', 'application/octet-stream', 'application/vnd.ms-excel', 'application/x-csv', 'text/x-csv', 'text/csv', 'application/csv', 'application/excel', 'application/vnd.msexcel', 'text/plain');
+
+
+		$file = new finfo(FILEINFO_MIME_TYPE);
+		$check = array_search($file->file($_FILES['file-upload']['tmp_name']), $csvMimes, true);
+
+		if ($check === false) {
+			throw new RuntimeException('Invalid file format.');
+		}
+
+		$file_csv = fopen($_FILES['file-upload']['tmp_name'], 'r');
+
+		if ( !$file_csv ) {
+			throw new RuntimeException("Could not open the file.");
+		}
+
+		while (($line = fgets($file_csv)) !== false) {
+			$line_array = explode(',', $line);
+
+			$test_id = $import->importTest($_SESSION["user_id"], $line_array[0]);
```

Comment:
There is possibility that `$_SESSION["user_id"]` is not set because the session has expired or a not logged user is trying to access the page. Please have a check for that up in the script.

Reply:
Added check if the user is logged in.